amundsen_application/api/utils/metadata_utils.py

```diff
@@ -1,73 +1,44 @@
-from typing import Dict
-from flask import current_app as app
+from typing import Any, Dict
 
-from amundsen_application.models.user import load_user, dump_user
+from amundsen_common.models.popular_table import PopularTable, PopularTableSchema
+from amundsen_common.models.table import Table, TableSchema
+from flask import current_app as app
 
 
-def marshall_table_partial(table: Dict) -> Dict:
+def marshall_table_partial(table_dict: Dict) -> Dict:
     """
     Forms a short version of a table Dict, with selected fields and an added 'key'
     :param table: Dict of partial table object
     :return: partial table Dict
 
     TODO - Unify data format returned by search and metadata.
     """
-    table_name = table.get('table_name', '')
-    schema_name = table.get('schema', '')
-    cluster = table.get('cluster', '')
-    db = table.get('database', '')
-    return {
-        'cluster': cluster,
-        'database': db,
-        'description': table.get('table_description', ''),
-        'key': '{0}://{1}.{2}/{3}'.format(db, cluster, schema_name, table_name),
-        'name': table_name,
-        'schema_name': schema_name,
-        'type': 'table',
-        'last_updated_epoch': table.get('last_updated_epoch', None),
-    }
+    schema = PopularTableSchema(strict=True)
+    # TODO: consider migrating to validate() instead of roundtripping
+    table: PopularTable = schema.load(table_dict).data
+    results = schema.dump(table).data
+    # TODO: fix popular tables to provide these? remove if we're not using them?
+    results['last_updated_timestamp'] = None
+    results['type'] = 'table'
+
+    return results
 
 
-def marshall_table_full(table: Dict) -> Dict:
+def marshall_table_full(table_dict: Dict) -> Dict:
     """
     Forms the full version of a table Dict, with additional and sanitized fields
     :param table: Table Dict from metadata service
     :return: Table Dict with sanitized fields
     """
-    # Filter and parse the response dictionary from the metadata service
-    fields = [
-        'badges',
-        'columns',
-        'cluster',
-        'database',
-        'is_view',
-        'key',
-        'owners',
-        'schema',
-        'source',
-        'table_description',
-        'table_name',
-        'table_readers',
-        'table_writer',
-        'tags',
-        'watermarks',
-        # 'last_updated_timestamp' Exists on the response from metadata but is not used.
-        # This should also be consolidated with 'last_updated_epoch' to have the same name and format.
-    ]
 
-    results = {field: table.get(field, None) for field in fields}
+    schema = TableSchema(strict=True)
+    # TODO: consider migrating to validate() instead of roundtripping
+    table: Table = schema.load(table_dict).data
+    results: Dict[str, Any] = schema.dump(table).data
```

Comment:
Same thing here. Ideally we would add some `id` to all resource types in the base schema.
```
    # TODO: Add the 'key' or 'id' to the Table schema
    results['key'] = '{0}://{1}.{2}/{3}'.format(table.db, table.cluster, table.schema_name, table.table_name)
```

Reply:
I had the start of that here: https://github.com/lyft/amundsencommon/pull/14/files but we'd need to add it to popular tables as well.